### PR TITLE
Create Views allowed for same keyspace

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR enables view support on information schema queries by restricting the create views to refer only single keyspace.
The views are now created at the MySQL level so all information schema, show query, describe, and explain query can work.

## Related Issue(s)

- https://github.com/vitessio/vitess/issues/11559

## Checklist

-   [X] Tests were added or are not required
-   [X] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/endtoend/views_test.go [hunk 1]
```diff
@@ -153,6 +152,77 @@ func TestDropViewDDL(t *testing.T) {
 	require.Zero(t, qr.Rows)
 }
 
+// TestViewDDLWithInfrSchema will validate information schema queries with views.
+func TestViewDDLWithInfrSchema(t *testing.T) {
```
Comment (frouioui) [inline go/vt/vttablet/endtoend/views_test.go:156]: We should perhaps add an E2E tests where we try to add a view with two different keyspaces in the select query and look for `VT12001` error

**Review by frouioui — APPROVED** [review]

File: go/vt/vttablet/endtoend/views_test.go (hunk 1 above)
Comment (harshit-gangal) [reply go/vt/vttablet/endtoend/views_test.go:156]: we have a unit test in plan_test that covers it.
views_test is vttablet e2e test, the failure will happen in vtgate while planning.

File: go/vt/vtgate/planbuilder/testdata/view_cases.json [hunk 2]
```diff
@@ -19,23 +19,9 @@
...
-          "Name": "user",
-          "Sharded": true
-        },
-        "Query": "create view view_a as select id from unsharded union select id from unsharded_auto union select id from unsharded_auto union select `name` from unsharded"
-      },
-      "TablesUsed": [
-        "user.view_a"
-      ]
-    }
+    "plan": "VT12001: unsupported: Select query does not belong to the same keyspace as the view statement"
```
Comment (GuptaManan100) [inline go/vt/vtgate/planbuilder/testdata/view_cases.json:24]: Maybe we should move this to unsupported cases now. There are people who use that list to see what all won't work with Vitess and this seems like a good addition to that.

File: go/vt/vttablet/tabletserver/health_streamer.go [hunk 3]
```diff
@@ -380,7 +382,13 @@ func getChangedTableNames(ctx context.Context, conn *connpool.DBConn) ([]string,
 	}
 	alloc := func() *sqltypes.Result { return &sqltypes.Result{} }
 	bufferSize := 1000
-	err := conn.Stream(ctx, mysql.DetectSchemaChange, callback, alloc, bufferSize, 0)
+
+	schemaChangeQuery := mysql.DetectSchemaChange
+	// If views are enabled, then views are tracked/handled separately and schema change does not need to track them.
+	if hs.viewsEnabled {
+		schemaChangeQuery = mysql.DetectSchemaChangeOnlyBaseTable
+	}
```
Comment (GuptaManan100) [inline go/vt/vttablet/tabletserver/health_streamer.go:390]: Can't we always use `mysql.DetectSchemaChangeOnlyBaseTable`?

**Review by GuptaManan100 — APPROVED** [review]

File: go/vt/vtgate/planbuilder/testdata/view_cases.json (hunk 2 above)
Comment (frouioui) [reply go/vt/vtgate/planbuilder/testdata/view_cases.json:24]: That's a good idea

File: go/vt/vttablet/tabletserver/health_streamer.go (hunk 3 above)
Comment (harshit-gangal) [reply go/vt/vttablet/tabletserver/health_streamer.go:390]: There is a possibility of users creating views on MySQL and providing that in vschema as table, so schema tracking should be able to populate column list for those when vitess managed views are not enabled.

File: go/vt/vtgate/planbuilder/testdata/view_cases.json (hunk 2 above)
Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/testdata/view_cases.json:24]: there is already a case in unsupported that fails with this error.

**Pull request merged**